### PR TITLE
Build a basic generic UI

### PR DESCRIPTION
**Pull request by akshatnitd**

### CODE REVIEW

**Comment by ghoshbishakh** [comment]
screenshot from 2017-01-27 21-46-22

File: festflow/static/css/main.css [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+.dash {
+	display: inline-block;
+	width: 100%;
+	background-color: #4285f4;
+	height: 5px;
+}
+
+ .jumbotron {
+ 	margin-top: 100px;
+ }
```
Comment (ghoshbishakh) [inline festflow/static/css/main.css]: Add newline

File: festflow/static/css/main.css [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	color: #FFFFFF;
+	text-transform: uppercase;
+	font-weight: 500;
+}
+
+.navbar a:hover {
+	color: #4285f4;
+}
+
+body {
```
Comment (ghoshbishakh) [inline festflow/static/css/main.css]: are you sure you want to put a background image in body?

File: festflow/static/css/main.css [hunk 3]
```diff
@@ -0,0 +1,65 @@
+.navbar {
+	background-color: #4285f4;
```
Comment (ghoshbishakh) [inline festflow/static/css/main.css]: Add proper comments to separate different css for different pages or even better use different css files for different views. And a common one.

**Comment by akshatnitd** [comment]
Okay sure, Will make the necessary changes.

File: festflow/templates/festflow/about.html [hunk 4]
```diff
@@ -1,13 +1,26 @@
+{% load static %}
+
 <!DOCTYPE html>
 <html>
 <head>
     <title>ABOUT PAGE</title>
+    <link rel="stylesheet" type="text/css" href='{% static "css/main.css" %}'/>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: first put meta tags. then link tags

File: festflow/templates/festflow/about.html [hunk 5]
```diff
@@ -1,13 +1,26 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+  	<link rel="stylesheet" type="text/css" href='{% static "css/bootstrap.min.css" %}'/>
+  	<link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: Please dwnload any font you want to use to static

File: festflow/templates/festflow/about.html [hunk 6]
```diff
@@ -1,13 +1,26 @@
...
+	<div class="container">
+		<div class="jumbotron">
+    		<h1>About Us</h1> 
+    		<div class="dash"><div>     
+    		<br />
+	    	<div class="about-text">
+		    	{% if content %}
+			        {{ content | safe }}
+			    {% endif %}
+			</div>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: Use proper indentation

File: festflow/templates/festflow/about.html [hunk 7]
```diff
@@ -1,13 +1,26 @@
...
+  		</div>
+	</div>
 </body>
 </html>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: newline

File: festflow/templates/festflow/index.html [hunk 8]
```diff
@@ -4,12 +4,23 @@
 <html>
 <head>
   <title>Home Page</title>
+  <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1" />
   <link rel="stylesheet" type="text/css" href='{% static "css/bootstrap.min.css" %}'/>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/index.html]: Use a comment to specify what stuff you are importing

File: festflow/templates/festflow/index.html [hunk 9]
```diff
@@ -4,12 +4,23 @@
...
+  <link href="https://fonts.googleapis.com/css?family=Roboto" rel="stylesheet">
+  <link rel="stylesheet" type="text/css" href='{% static "css/main.css" %}'/>
 </head>
 <body>
   {% navbar %}
-  <h1>Home Page</h1>
+  <div class="title">
+  	<div class="logo">
+  		<a href="#home">
+  			<img src="{% static "images/logo.png" %}" alt="Logo Brand">
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/index.html]: use static font

File: festflow/templates/festflow/index.html [hunk 10]
```diff
@@ -4,12 +4,23 @@
...
+  		</a>
+  	</div>
+  	<h1>FEST TITLE</h1>
+  	<h4>A one-liner description of the fest</h4>
+  <div>
 </body>
 <script type="text/javascript" href='{% static "js/jquery.min.js"%}'></script>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/index.html]: add comment to specify what stuff you are importing

File: festflow/templates/festflow/navbar.html [hunk 11]
```diff
@@ -1,12 +1,25 @@
...
+      </ul>
     </div>
-    <ul class="nav navbar-nav navbar-right">
-      {% for entry in entries %}
-      <li><a href="{{ entry.url }}">{{ entry.name }}</a></li>
-      {% endfor %}
-    </ul>
   </div>
-</nav>
+</nav>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/navbar.html]: newline

**Comment by akshatnitd** [comment]
How does this look?

**Comment by ghoshbishakh** [comment]
@akshatnitd looks good

**Comment by ghoshbishakh** [comment]
@akshatnitd do not change the PR. Push in this one. Make the changes. Commit and squash.

**Comment by akshatnitd** [comment]
Made the changes (Y)

File: festflow/static/css/about.css [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+	color: #4d4d4d;
+}
+
+.dash {
+	display: inline-block;
+	width: 100%;
+	background-color: #4285f4;
+	height: 5px;
+}
+
```
Comment (ghoshbishakh) [inline festflow/static/css/about.css]: newline

File: festflow/templates/festflow/about.html [hunk 13]
```diff
@@ -1,13 +1,34 @@
+{% load static %}
+
 <!DOCTYPE html>
 <html>
 <head>
-    <title>ABOUT PAGE</title>
+	<title>ABOUT PAGE</title>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: indent properly

File: festflow/templates/festflow/about.html [hunk 14]
```diff
@@ -1,13 +1,34 @@
...
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <!-- LOAD CSS AND BASIC JS -->
+
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: no reason to indent after comments

File: festflow/templates/festflow/index.html [hunk 15]
```diff
@@ -4,13 +4,31 @@
 <html>
 <head>
   <title>Home Page</title>
+  <meta charset="utf-8">
   <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <!-- LOAD CSS AND BASIC JS -->
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/index.html]: use uniform indentation. either 4 spaces or a tab or 2 spaces

File: festflow/templates/festflow/about.html (hunk 13 above)
Comment (nkprince007) [inline festflow/templates/festflow/about.html]: Just two spaces please, because, all other markup files use the similar indents.

**Comment by akshatnitd** [comment]
Have made the changes. Let me know if any other changes to be made. Will be working on the other views now.

File: festflow/static/css/main.css [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+body{
+	background: url('../images/bg.png');
+}
+
+/*Navbar styling*/
+.navbar {
+	background-color: #4285f4;
+	text-decoration: none;
+	-webkit-box-shadow: 5px 5px 5px #aaa;
+    -moz-box-shadow: 5px 5px 5px #aaa;
```
Comment (nkprince007) [inline festflow/static/css/main.css]: messed up indentation here.

File: festflow/templates/festflow/index.html [hunk 17]
```diff
@@ -4,13 +4,31 @@
...
+  <!-- Main CSS -->
+  <link rel="stylesheet" type="text/css" href='{% static "css/main.css" %}'/>
+
 </head>
 <body>
   {% navbar %}
-  <h1>Home Page</h1>
+  <div class="title">
+    <div class="logo">
+  		<img src="{% static "images/logo.png" %}" alt="Logo Brand">
```
Comment (nkprince007) [inline festflow/templates/festflow/index.html]: another messed up indentation here.

**Comment by ghoshbishakh** [comment]
@akshatnitd In about page there is no navbar. Also images do not have a max width.
screenshot from 2017-01-28 11-30-54
screenshot from 2017-01-28 11-31-56

**Comment by akshatnitd** [comment]
@nkprince007 @ghoshbishakh Had some issues with the text editor. Fixed the indent issues. Added navbar to the about page and added max-width to the images.

**Comment by ghoshbishakh** [comment]
@akshatnitd you have not made any changes. Please push

**Comment by akshatnitd** [comment]
@ghoshbishakh Doing that. Should I also set any max-height for the image inside the about container?

**Comment by ghoshbishakh** [comment]
@akshatnitd no maxheight is meaningless. Just use `max-width: 100%`

**Comment by akshatnitd** [comment]
@ghoshbishakh Done the changes.

**Comment by ghoshbishakh** [comment]
screenshot from 2017-01-28 15-02-23


@akshatnitd too much margin top in the about page. Reduce the gap between nav and the content.

And one bug: In mobile view the nav links disappear and no dropdown appears.

File: festflow/templates/festflow/about.html [hunk 18]
```diff
@@ -1,13 +1,34 @@
...
 </head>
 <body>
-    {% if content %}
-        {{ content | safe }}
-    {% else %}
-        ABOUT PAGE
-    {% endif %}
+  <div class="container">
+    <div class="jumbotron">
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/about.html]: Do not use jumbotron. Use row and a column

**Comment by akshatnitd** [comment]
Will resolve these issues :)

File: festflow/templates/festflow/event_view.html [hunk 19]
```diff
@@ -1,29 +1,67 @@
...
+			{% if event.logo %}
+			<img src="{{ event.logo.url }}" alt= "{{ event.name }}" class='img-responsive' />
+			<br />
+			{% endif %}
+
+			<h3 id='link'><strong>About the Event</strong></h3>
+			<div class="subdash"></div>
+			<br />
+			<br />
+			<p>{{ event.description }}</p>
```
Comment (ghoshbishakh) [inline festflow/templates/festflow/event_view.html]: use {{ event.description|safe }}

Comment (akshatnitd) [inline festflow/templates/festflow/event_view.html]: Will make the change.

**Comment by ghoshbishakh** [comment]
@akshatnitd Login page has issues. Plase make sure buttons resize properly.
screenshot from 2017-01-29 12-36-13

**Comment by ghoshbishakh** [comment]
TODO (do it later)
@akshatnitd In the events page try to use a library like http://masonry.desandro.com/

That will fix the spacing issue.
screenshot from 2017-01-29 12-46-49

**Comment by nkprince007** [comment]
Masonry is awesome, I've used it on muZic-electron(one of my repos).. 

@ghoshbishakh @akshatnitd 👍 for masonry grid.

**Comment by ghoshbishakh** [comment]
@akshatnitd Looks like you are not using template inheritance. It is really simple. Just go through this post: https://tutorial.djangogirls.org/en/template_extending/

Now create a `base.html` that will contain all css and js imports and navbar and title and meta tags.
Then just extend the body part.

**Comment by akshatnitd** [comment]
@ghoshbishakh Will make the changes, and will implement the mansory grid after all the pages are designed :+1:

**Comment by akshatnitd** [comment]
@ghoshbishakh Checking the linting issues :+1:

**Comment by ghoshbishakh** [comment]
@akshatnitd excellend job with the linting! Keep it up

**Comment by akshatnitd** [comment]
@ghoshbishakh Cool :+1: Working on other templates now!

**Comment by ghoshbishakh** [comment]
@akshatnitd remember tomorrow is your submisssion deadline
its Noon 12. So you have to submit it by morning

**Comment by ghoshbishakh** [comment]
@akshatnitd You have probably specified some larger font for paragraphs in events_view page. As a result texts in lists appear small. Please check.
screenshot from 2017-01-30 23-24-29

**Comment by akshatnitd** [comment]
@ghoshbishakh Made the change, not all tags under jumbotron are of same size !
Also pushed the code for other templates.

**Comment by ghoshbishakh** [comment]
@akshatnitd please change the last commit message. Be specific about what remaing templates

**Comment by ghoshbishakh** [comment]
@akshatnitd atleast remove the word "remaining".

File: festflow/templates/404.html [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+{% block body_contents %}
+
+<div class="container">
+    <div class="title">
+        <div class="logo">
+            <img src="{% static "images/logo.png" %}" class="img-responsive" alt="Logo Brand">
+        </div>
+        <h1>404 ERROR</h1>
+        <h4>This page isn't available. Sorry about that.</h4>
```
Comment (ghoshbishakh) [inline festflow/templates/404.html]: @akshatnitd  Please change this text to something like 'The page you are looking for does not exist.'

File: festflow/templates/500.html [hunk 21]
```diff
@@ -0,0 +1,16 @@
...
+
+{% block body_contents %}
+
+<div class="container">
+    <div class="title">
+        <div class="logo">
+            <img src="{% static "images/logo.png" %}" class="img-responsive" alt="Logo Brand">
+        </div>
+        <h1>500 ERROR</h1>
+        <h4>This page isn't available. Sorry about that.</h4>
```
Comment (ghoshbishakh) [inline festflow/templates/500.html]: @akshatnitd please change the text to something like.. 'Opps something went wrong.. Internal Server Error'

**Comment by ghoshbishakh** [comment]
@akshatnitd Submission LINK: 
https://docs.google.com/forms/d/e/1FAIpQLScC9f0OZVQUM2pfZZ7MLzs4Z1JvfzK25oKDXChhEC1yrWVPVQ/viewform?c=0&w=1

Please share the gist with me before submitting.

File: festflow/templates/404.html (hunk 20 above)
Comment (akshatnitd) [inline festflow/templates/404.html]: Okay sure :)

File: festflow/templates/500.html (hunk 21 above)
Comment (akshatnitd) [inline festflow/templates/500.html]: Okay sure :)

**Comment by akshatnitd** [comment]
@ghoshbishakh Thanks for the submission link! Will share before submitting.

**Comment by akshatnitd** [comment]
Made the final commit for error message change on 404.html and 500.html.

And glad to see the green ticks :smile: :smiley:

**Comment by ghoshbishakh** [comment]
@akshatnitd I see you have included masonry but not used it? You can do it in a separate PR.
@nkprince007 please check this and confirm merge!

**Comment by akshatnitd** [comment]
@ghoshbishakh I was just trying to use it.
Will do it in separate PR.
Should I remove the mansory static files?

**Comment by ghoshbishakh** [comment]
@akshatnitd no let it be. Let this pr be merged first

**Comment by akshatnitd** [comment]
Okay.. Will do that part later.

ᐧ

**Comment by nkprince007** [comment]
Looks good to me @ghoshbishakh

**Pull request merged**